tasks/definevscvm.yml

```diff
@@ -0,0 +1,37 @@
+# List running VM
+- name: Get a list of VMs
+  virt: command=list_vms
+  register: virt_vms
+  delegate_to: "{{ hypervisor }}"
+
+# Destroy if running VSC Guest VM
+- name: Destroy VSC Guest VM if still running
+  virt: name={{ inventory_hostname }} state=destroyed
+  delegate_to: "{{ hypervisor }}"
+  when: inventory_hostname in virt_vms.list_vms  
+
+# Undefine any previously existing VSC Guest VM
+- name: Undefine VSC Guest VM if already provisioned
+  virt: name={{ inventory_hostname }} command=undefine
+  delegate_to: "{{ hypervisor }}"
+  when: inventory_hostname in virt_vms.list_vms
+
+- name: Copy over VSC qcow2 image
```

Comment:
The first copy is the "src" one.
This is to do a comparison if the binary has changed. This is the case when
you want to install a newer version. It is impossible to compare this on
the actual qcow since this has a changed config file inside.

The second copy will be the actual qcow where the config has to be embbeded
into.


A complete different alternative implementation is to install netmiko
library and request the version dynamically.




On Tue, May 24, 2016 at 9:42 PM, Roman Dodin <notifications@github.com>
wrote:

> In tasks/definevscvm.yml
> <https://github.com/rvichery/nuage-vsc-installer/pull/6#discussion_r64456444>
> :
>
> > +  register: virt_vms
> > +  delegate_to: "{{ hypervisor }}"
> > +
> > +# Destroy if running VSC Guest VM
> > +- name: Destroy VSC Guest VM if still running
> > +  virt: name={{ inventory_hostname }} state=destroyed
> > +  delegate_to: "{{ hypervisor }}"
> > +  when: inventory_hostname in virt_vms.list_vms
> > +
> > +# Undefine any previously existing VSC Guest VM
> > +- name: Undefine VSC Guest VM if already provisioned
> > +  virt: name={{ inventory_hostname }} command=undefine
> > +  delegate_to: "{{ hypervisor }}"
> > +  when: inventory_hostname in virt_vms.list_vms
> > +
> > +- name: Copy over VSC qcow2 image
>
> It seems like another copying procedure. The first one is still in place
> in main.yml
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/rvichery/nuage-vsc-installer/pull/6/files/51d3faa272440df5112ebcfc6f4eb001311e097e#r64456444>
>